test/ffmpeg-vaapi/vpp/tonemap.py

```diff
@@ -0,0 +1,46 @@
+###
+### Copyright (C) 2018-2023 Intel Corporation
+###
+### SPDX-License-Identifier: BSD-3-Clause
+###
+
+from ....lib import *
+from ....lib.ffmpeg.vaapi.util import *
+from ....lib.ffmpeg.vaapi.vpp import VppTest
+
+@slash.requires(*platform.have_caps("vpp", "tonemap"))
+@slash.requires(*have_ffmpeg_filter("tonemap_vaapi"))
+class TonemapTest(VppTest):
+  def before(self):
+    vars(self).update(
+      vpp_op  = "tonemap",
+    )
+    super().before()
+
+  def init(self, tspec, case, mode, csc):
+    vars(self).update(tspec[case].copy())
+    vars(self).update(case = case, mode = mode, csc = csc)
+
+  def validate_caps(self):
+    self.caps = platform.get_caps("vpp", "tonemap", self.mode)
+
+    if self.caps is None:
+      slash.skip_test(
+        format_value(
+          "{platform}.{driver}.{mode} not supported", **vars(self)))
+
+    super().validate_caps()
+
+spec_hevc = load_test_spec("vpp", "tonemap", "hevc_10")
+@slash.requires(*platform.have_caps("decode", "hevc_10"))
+@slash.requires(*have_ffmpeg_filter_options("tonemap_vaapi", "format"))
+@slash.requires(*have_ffmpeg_decoder("hevc"))
+class hevc(TonemapTest):
+  def before(self):
+    self.ffdecoder = "hevc"
+    super(hevc, self).before()
+
+  @slash.parametrize(*gen_vpp_h2s_parameters(spec_hevc))
+  def test_h2s(self, case, mode, csc):
```

Comment:
Thanks for the comments. I removed the `validate_caps` override and  the `mode` parameter from the test function signature in the new version.  Because `h2h` mode requires more parameters, I prefer to use separate param generator and test function for each mode.